src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1653,9 +1657,9 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                         // also run the nullable walker, and issue duplicate warnings. We should try to only run the pass
                         // once.
                         // https://github.com/dotnet/roslyn/issues/35041
-                        methodBody = NullableWalker.AnalyzeAndRewrite(bodyBinder.Compilation, method, methodBody, bodyBinder, new DiagnosticBag(), createSnapshots: false, out _);
+                        methodBody = NullableWalker.AnalyzeAndRewrite(bodyBinder.Compilation, method, methodBody, bodyBinder, new DiagnosticBag(), createSnapshots: true, out snapshotManager);
```

Comment:
>methodBody = NullableWalker.AnalyzeAndRewrite(bodyBinder.Compilation, method, methodBody, bodyBinder, new DiagnosticBag(), createSnapshots: true, out snapshotManager); [](start = 24, length = 167)

This isn't introduced in this PR, but it looks like we are doing a lot of extra work even when no-one needs a SemanticModel. Consider opening an issue to delay this work until someone actually starts using SemanticModel.

Reply:
Added a note to [https://github.com/dotnet/roslyn/issues/35041](https://github.com/dotnet/roslyn/issues/35041).

---
In reply to: [295516436](https://github.com/dotnet/roslyn/pull/36563#discussion_r295516436) [](ancestors = 295516436)